### PR TITLE
Variables as symbols

### PR DESCRIPTION
**Pull request by schymans**
Refactored Variables to be based on Symbol instead of Quantity. This solves Issue #38 to some extent and eliminates the need for `._name` whenever a symbolic expression of a variable is needed.

### CODE REVIEW

**Comment by schymans** [comment]
This is strange, travis says: `check-manifest
lists of files in version control and sdist match`
But locally, I get:
```
check-manifest -u -v
listing source files under version control: 60 files and directories
building an sdist: essm-1.2.0.dev20180323.tar.gz: 60 files and directories
copying source files to a temporary directory
building a clean sdist: essm-1.2.0.dev20180323.tar.gz: 60 files and directories
lists of files in version control and sdist match
```

**Comment by jirikuncar** [comment]
@schymans can you add a test for:

```python
class a(Variable):
    unit = meter

class b(Variable):
    unit = second

with pytest.raises(Exception):
    result = a + b
```

**Comment by schymans** [comment]
I think it is OK if a + b does not raise an error, unless it is part of a definition in Variable() or Equation(). This way, we can use variables just as any symbols and perform unit checking on an expression whenever we like, using `Variable.check_unit(expr)`. Otherwise we would need to follow an approach as suggested here: https://github.com/sympy/sympy/issues/14318#issuecomment-369583972

File: essm/variables/units.py [hunk 1]
```diff
@@ -61,8 +61,14 @@ def derive_quantity(expr, name=None):
 
 def derive_unit(expr, name=None):
     """Derive SI-unit from an expression, omitting scale factors."""
+    from essm.variables.utils import extract_variables
+    variables = extract_variables(expr)
+    dict_units = {}
+    for variable in variables:
+        dict_units[variable] = variable.definition.unit
+    expr_units = expr.xreplace(dict_units)
```
Comment (jirikuncar) [inline essm/variables/units.py]: Have you tried: `expr.xreplace(Variable.__units__)`?

File: essm/variables/utils.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 from sympy import Eq, preorder_traversal
 from sympy.core.expr import Expr
+from sympy.physics.units import Dimension
```
Comment (jirikuncar) [inline essm/variables/utils.py]: do you need this import?

File: essm/variables/units.py (hunk 1 above)
Comment (schymans) [reply essm/variables/units.py]: Yes, but I cannot import Variable into units.py, probably due to circular import issues.

File: essm/variables/utils.py (hunk 2 above)
Comment (schymans) [reply essm/variables/utils.py]: Thanks, I just removed it.

File: essm/variables/units.py (hunk 1 above)
Comment (schymans) [reply essm/variables/units.py]: OK, thanks for the tip to import Variable within the function. Fixed.

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**